hs-abci-sdk/src/Tendermint/SDK/BaseApp/Transaction/Effect.hs

```diff
@@ -95,11 +100,12 @@ runTx ps ctx@TransactionContext{..} tx = do
         def & txResultGasWanted .~ G.unGasAmount initialGas
             & txResultGasUsed .~ G.unGasAmount gasUsed
   case eRes of
-    Left e -> return (baseResponse & txResultAppError .~ e, Nothing)
+    Left e -> return (Nothing, baseResponse & txResultAppError .~ e, Nothing)
     Right a -> do
         es <- liftIO $ readIORef events
         c <- liftIO $ readIORef storeCache
-        return ( baseResponse & txResultEvents .~ es
+        return ( Just a
```

Comment:
It's probably better to use `(Maybe (a, Cache.Cache), TxResult)` as the return type since `(Just _, _, Nothing)` and `(Nothing, _, Just)` are both impossible

Reply:
Agreed, better represents the actual results.